core/node/libp2p/rcmgr_defaults.go

```diff
@@ -18,63 +18,64 @@ import (
-	if minOutbound := 65536; defaultLimits.SystemBaseLimit.ConnsOutbound < minOutbound {
-		defaultLimits.SystemBaseLimit.ConnsOutbound = minOutbound
-	}
-	if minFD := 4096; defaultLimits.SystemBaseLimit.FD < minFD {
-		defaultLimits.SystemBaseLimit.FD = minFD
-	}
-
+	// // Adjust limits
+	// // (based on https://github.com/filecoin-project/lotus/pull/8318/files)
+	// // - give it more memory, up to 4G, min of 1G
+	// // - if Swarm.ConnMgr.HighWater is too high, adjust Conn/FD/Stream limits
+	// defaultLimits := rcmgr.DefaultLimits.WithSystemMemory(.125, 1<<30, 4<<30)
+	//
+	// // Outbound conns and FDs are set very high to allow for the accelerated DHT client to (re)load its routing table.
+	// // Currently it doesn't gracefully handle RM throttling--once it does we can lower these.
+	// // High outbound conn limits are considered less of a DoS risk than high inbound conn limits.
+	// // Also note that, due to the behavior of the accelerated DHT client, we don't need many streams, just conns.
+	// if minOutbound := 65536; defaultLimits.SystemBaseLimit.ConnsOutbound < minOutbound {
+	// 	defaultLimits.SystemBaseLimit.ConnsOutbound = minOutbound
+	// }
+	// if minFD := 4096; defaultLimits.SystemBaseLimit.FD < minFD {
+	// 	defaultLimits.SystemBaseLimit.FD = minFD
+	// }
+	//
 	// Do we need to adjust due to Swarm.ConnMgr.HighWater?
-	if cfg.ConnMgr.Type == "basic" {
-		maxconns := cfg.ConnMgr.HighWater
-		if 2*maxconns > defaultLimits.SystemBaseLimit.ConnsInbound {
-			// Conns should be at least 2x larger than the high water to allow for two conns per peer (TCP+QUIC).
-			defaultLimits.SystemBaseLimit.ConnsInbound = logScale(2 * maxconns)
-
-			// We want the floor of minOutbound conns to be no less than what was set above.
-			if minOutbound := logScale(2 * maxconns); minOutbound > defaultLimits.SystemBaseLimit.ConnsOutbound {
-				defaultLimits.SystemBaseLimit.ConnsOutbound = minOutbound
-			}
-
-			if 2*maxconns > defaultLimits.SystemBaseLimit.FD {
-				defaultLimits.SystemBaseLimit.FD = logScale(2 * maxconns)
-			}
-
-			defaultLimits.SystemBaseLimit.StreamsInbound = logScale(16 * maxconns)
-			defaultLimits.SystemBaseLimit.StreamsOutbound = logScale(64 * maxconns)
-			defaultLimits.SystemBaseLimit.Streams = logScale(64 * maxconns)
-
-			defaultLimits.ServiceBaseLimit.StreamsInbound = logScale(8 * maxconns)
-			defaultLimits.ServiceBaseLimit.StreamsOutbound = logScale(32 * maxconns)
-			defaultLimits.ServiceBaseLimit.Streams = logScale(32 * maxconns)
-
-			defaultLimits.ProtocolBaseLimit.StreamsInbound = logScale(8 * maxconns)
-			defaultLimits.ProtocolBaseLimit.StreamsOutbound = logScale(32 * maxconns)
-			defaultLimits.ProtocolBaseLimit.Streams = logScale(32 * maxconns)
-		}
-	}
-
-	defaultLimits.SystemBaseLimit.Conns = defaultLimits.SystemBaseLimit.ConnsOutbound + defaultLimits.SystemBaseLimit.ConnsInbound
-
-	return defaultLimits
+	// if cfg.ConnMgr.Type == "basic" {
```

Comment:
This was weird. I would expect the resource manager be the ultimate arbiter, not connmgr.

Reply:
We need something analogous though. The connection manager limit needs to be in sync with resource manager limit.

Reply:
If these two are out of sync which should be the source of truth?

Reply:
Probably the rcmgr, just because it's configuration is so much more complex. If I remember correctly, there were plans to deprecate the connmgr watermark configuration options in go-ipfs.